### ISSUE TITLE
How to set pipe "requires_grad=true"？

### ISSUE THREAD

**Issue by micklexqg** [huggingface/diffusers#12672]
I have set the variable and the model "requires_grad=true" with the following:
`  pipe.transformer.requires_grad = True
    pipe.vae.requires_grad = True`
`prev_sample = prev_sample.detach().requires_grad_(True)`
but the "requires_grad" of result by the pipe is still not true:
   `image_tar = pipe.vae.decode(prev_sample, return_dict=False)[0]`
"image_tar" still can  not requires_grad, so how to set pipe "requires_grad=true"？(all the operation is during inference stage.)

**Comment by DN6** [comment]
The pipeline __call__ method uses a `torch.no_grad` decorator. 
https://github.com/huggingface/diffusers/blob/b7df4a5387fd65edf4f16fd5fc8c87a7c815a4c7/src/diffusers/pipelines/flux/pipeline_flux.py#L652

Pipelines don't set requires grad since they are designed for inference. You could override it with something like

```python
  from diffusers import FluxPipeline

  class GradientEnabledPipeline(FluxPipeline):
      def __call__(self, ...):
          return super().__call__(...)

```

**Comment by micklexqg** [comment]
> The pipeline **call** method uses a `torch.no_grad` decorator.
> 
> [diffusers/src/diffusers/pipelines/flux/pipeline_flux.py](https://github.com/huggingface/diffusers/blob/b7df4a5387fd65edf4f16fd5fc8c87a7c815a4c7/src/diffusers/pipelines/flux/pipeline_flux.py#L652)
> 
> Line 652 in [b7df4a5](/huggingface/diffusers/commit/b7df4a5387fd65edf4f16fd5fc8c87a7c815a4c7)
> 
>  @torch.no_grad() 
> Pipelines don't set requires grad since they are designed for inference. You could override it with something like
> 
>   from diffusers import FluxPipeline
> 
>   class GradientEnabledPipeline(FluxPipeline):
>       def __call__(self, ...):
>           return super().__call__(...)

@DN6 ，thanks, I forgot there is   "@torch.no_grad()" and all is ok after removing it.
